projects/ngneat/cashew/src/lib/cache-context.ts

```diff
@@ -16,8 +16,8 @@ export interface ContextOptions {
 
 export const CACHE_CONTEXT = new HttpContextToken<ContextOptions>(() => ({}));
 
-export function withCache(options: ContextOptions = {}) {
-  return new HttpContext().set(CACHE_CONTEXT, {
+export function withCache(options: ContextOptions = {}, existingHttpContext?: HttpContext) {
```

Comment:
I prefer to add it as an option. I don't like the `withCache({}, context)` style.

Take it from the options and delete it afterward.